[workflows] CI macOS build failing in CMake configuration step

The configuration step is failing with the following error. (See https://github.com/robotology/whole-body-estimators/runs/557385561?check_suite_focus=true)

``` bash
2020-04-03T08:28:12.3307690Z -- Found YARP: /Users/runner/runners/2.168.0/work/whole-body-estimators/whole-body-estimators/install/deps/lib/cmake/YARP (found version "3.3.2")
2020-04-03T08:28:12.3420550Z -- Using iCub from install
2020-04-03T08:28:12.3838150Z -- Found YARP: /Users/runner/runners/2.168.0/work/whole-body-estimators/whole-body-estimators/install/deps/lib/cmake/YARP (found version "3.3.2")
2020-04-03T08:28:12.3839920Z CMake Error at install/deps/lib/ICUB/icub-config.cmake:38 (include):
2020-04-03T08:28:12.3840140Z   include could not find load file:
2020-04-03T08:28:12.3840200Z 
2020-04-03T08:28:12.3840780Z     /Users/runner/runners/2.165.2/work/whole-body-estimators/whole-body-estimators/install/deps/lib/ICUB/icub-export-install.cmake
2020-04-03T08:28:12.3840950Z Call Stack (most recent call first):
2020-04-03T08:28:12.3841080Z   /usr/local/Cellar/cmake/3.17.0_1/share/cmake/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
2020-04-03T08:28:12.3841220Z   install/deps/lib/cmake/iDynTree/iDynTreeConfig.cmake:23 (find_dependency)
2020-04-03T08:28:12.3841870Z   devices/baseEstimatorV1/CMakeLists.txt:7 (find_package)
2020-04-03T08:28:12.3841930Z 
2020-04-03T08:28:12.3842000Z 
2020-04-03T08:28:12.3842470Z CMake Error at install/deps/lib/ICUB/icub-config.cmake:39 (include):
2020-04-03T08:28:12.3842630Z   include could not find load file:
2020-04-03T08:28:12.3842710Z 
2020-04-03T08:28:12.3843250Z     /Users/runner/runners/2.165.2/work/whole-body-estimators/whole-body-estimators/install/deps/lib/ICUB/icub-export-inst-includes.cmake
2020-04-03T08:28:12.3843370Z Call Stack (most recent call first):
2020-04-03T08:28:12.3843510Z   /usr/local/Cellar/cmake/3.17.0_1/share/cmake/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
2020-04-03T08:28:12.3843640Z   install/deps/lib/cmake/iDynTree/iDynTreeConfig.cmake:23 (find_dependency)
2020-04-03T08:28:12.3843770Z   devices/baseEstimatorV1/CMakeLists.txt:7 (find_package)
2020-04-03T08:28:12.3843860Z 
2020-04-03T08:28:12.3843900Z 
2020-04-03T08:28:12.4023810Z --  [x] plugin baseEstimatorV1 (ENABLE_baseEstimatorV1)
2020-04-03T08:28:12.5097390Z -- Found YARP: /Users/runner/runners/2.168.0/work/whole-body-estimators/whole-body-estimators/install/deps/lib/cmake/YARP (found version "3.3.2")
2020-04-03T08:28:12.5101440Z -- Using iCub from install
2020-04-03T08:28:12.5510300Z -- Found YARP: /Users/runner/runners/2.168.0/work/whole-body-estimators/whole-body-estimators/install/deps/lib/cmake/YARP (found version "3.3.2")
2020-04-03T08:28:12.5512340Z CMake Error at install/deps/lib/ICUB/icub-config.cmake:38 (include):
2020-04-03T08:28:12.5512590Z   include could not find load file:
2020-04-03T08:28:12.5512680Z 
2020-04-03T08:28:12.5513190Z     /Users/runner/runners/2.165.2/work/whole-body-estimators/whole-body-estimators/install/deps/lib/ICUB/icub-export-install.cmake
2020-04-03T08:28:12.5513350Z Call Stack (most recent call first):
2020-04-03T08:28:12.5513470Z   devices/wholeBodyDynamics/CMakeLists.txt:7 (find_package)
2020-04-03T08:28:12.5513560Z 
2020-04-03T08:28:12.5513620Z 
2020-04-03T08:28:12.5514100Z CMake Error at install/deps/lib/ICUB/icub-config.cmake:39 (include):
2020-04-03T08:28:12.5514240Z   include could not find load file:
2020-04-03T08:28:12.5514290Z 
2020-04-03T08:28:12.5514800Z     /Users/runner/runners/2.165.2/work/whole-body-estimators/whole-body-estimators/install/deps/lib/ICUB/icub-export-inst-includes.cmake
2020-04-03T08:28:12.5514960Z Call Stack (most recent call first):
2020-04-03T08:28:12.5515090Z   devices/wholeBodyDynamics/CMakeLists.txt:7 (find_package)
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -83,7 +83,8 @@ jobs:
           uses: actions/cache@v1
           with:
             path: ${{ github.workspace }}/install/deps
-            key: source-deps-${{runner.os}}-vcpkg-robotology-${{env.vcpkg_robotology_TAG}}-ycm-${{env.YCM_TAG}}-yarp-${{env.YARP_TAG}}-icub-main-${{env.ICUB_TAG}}-iDynTree-${{env.iDynTree_TAG}}
+            # Including ${{ runner.temp }} is a workaround for https://github.com/robotology/whole-body-estimators/issues/60
+            key: source-deps-${{runner.os}}-${{runner.temp}}-vcpkg-robotology-${{env.vcpkg_robotology_TAG}}-ycm-${{env.YCM_TAG}}-yarp-${{env.YARP_TAG}}-icub-main-${{env.ICUB_TAG}}-iDynTree-${{env.iDynTree_TAG}}
                                                         
         - name: Source-based Dependencies [Windows]
           if: steps.cache-source-deps.outputs.cache-hit != 'true' && matrix.os == 'windows-latest'

